packages/jest-cli/src/lib/transform.js

```diff
@@ -17,13 +21,22 @@ const cache = new Map();
 const configToJsonMap = new Map();
 const preprocessorRegExpCache = new WeakMap();
 
+export type Processor = {
+  process: (sourceText: string, sourcePath: string) => string,
```

Comment:
Further down the file there's a check for preprocessor.process, and throws if it's not a function.